### PR TITLE
MGMT-12591:setting lvmo on non-SNO cluster should return bad request

### PR DESCRIPTION
**Pull request by subhamkrai**
setting lvmo on non-SNO cluster should return bad request

Signed-off-by: subhamkrai <srai@redhat.com>

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by gamli75 — COMMENTED** [review]
Why did you add the code in the LVM operator code? I think you should add it to https://github.com/openshift/assisted-service/blob/master/internal/bminventory/inventory.go#L580

**Comment by subhamkrai** [comment]
> Why did you add the code in the LVM operator code? I think you should add it to https://github.com/openshift/assisted-service/blob/master/internal/bminventory/inventory.go#L580

hmm, I think the suggestion makes more sense and it will change in one place, my changes were also working but it was returning an API error, it was just returning an empty operator list.

File: internal/operators/manager.go [hunk 1]
```diff
@@ -437,6 +438,11 @@ func EnsureLVMAndCNVDoNotClash(openshiftVersion string, operators []*models.Moni
 			cnvEnabled = true
 		}
 	}
+
+	if lvmEnabled && !common.IsSingleNodeCluster(cluster) {
+		return errors.Errorf("ODF LVM operator is only supported for Single Node Openshift")
```
Comment (gamli75) [inline internal/operators/manager.go:443]:
```suggestion
		return errors.Errorf("OpenShift Data Foundation Logical Volume Manager operator is only supported for Single Node Openshift")
```

File: internal/operators/manager.go [hunk 2]
```diff
@@ -44,6 +44,7 @@ type Manager struct {
 }
 
 // API defines Operator management operation
+//
```
Comment (gamli75) [inline internal/operators/manager.go:47]: please remove it

File: internal/operators/lvm/lvm_operator.go [hunk 3]
```diff
@@ -121,7 +121,7 @@ func (o *operator) ValidateHost(ctx context.Context, cluster *common.Cluster, ho
 }
 
 // GenerateManifests generates manifests for the operator
-func (o *operator) GenerateManifests(_ *common.Cluster) (map[string][]byte, []byte, error) {
+func (o *operator) GenerateManifests(cluster *common.Cluster) (map[string][]byte, []byte, error) {
```
Comment (gamli75) [inline internal/operators/lvm/lvm_operator.go:124]: please revert this change

Comment (subhamkrai) [reply internal/operators/lvm/lvm_operator.go]: done

File: internal/operators/manager.go (hunk 2 above)
Comment (subhamkrai) [reply internal/operators/manager.go]: done

File: internal/operators/manager.go (hunk 1 above)
Comment (subhamkrai) [reply internal/operators/manager.go]: done

**Comment by subhamkrai** [comment]
I have tested the changes it is working 

Screenshot from 2022-12-14 15-41-52

**Review by akalenyu — COMMENTED** [review]
Maybe worth also to add some source about LVMO/topolvm not playing nice with non-SNO installs (docs/repo readme?)

**Comment by subhamkrai** [comment]
> Maybe worth also to add some source about LVMO/topolvm not playing nice with non-SNO installs (docs/repo readme?) Just for future reference on the PR

not able to find upstream doc, but here is downstream doc https://access.redhat.com/documentation/en-us/red_hat_openshift_data_foundation/4.10/html/deploying_openshift_data_foundation_on_single_node_openshift_clusters/installing-odf-logical-volume-manager-operator-using-rhacm_sno

**Comment by gamli75** [comment]
@subhamkrai please fix the unit test - test Ensure CNV or LVM enabled failed

**Review by gamli75 — APPROVED** [review]
